tree-sitter-stack-graphs/src/bin/tree-sitter-stack-graphs/test.rs

```diff
@@ -364,8 +381,14 @@ impl Command {
         Ok(())
     }
 
-    fn save_paths(&self, path: &Path, paths: &mut Paths, graph: &StackGraph) -> anyhow::Result<()> {
-        let json = paths.to_json(graph, |_, _, _| true).to_string_pretty()?;
+    fn save_paths(
+        &self,
+        path: &Path,
+        paths: &mut Paths,
+        graph: &StackGraph,
+        filter: &dyn Filter,
```

Comment:
Can you avoid the `&dyn Filter` here and just use `&Filter`, or am I misunderstanding something about Rust traits?

Reply:
Compiler says you should use `dyn` when passing traits around. It indicates that some dynamic dispatching is involved, but I don't know how this works under the hood, TBH.